lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,208 @@
+
+		$this->appName = $AppName;
+		// $this->initialStateService = $initialStateService;
+		$this->languageFactory = $languageFactory;
+		$this->manager = $manager;
+		$this->config = $config;
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * generate download url for a social entry (based on type of data requested)
+	 */
+	protected function getSocialConnector(array $socialentry, string $type) : ?string {
+		foreach ($socialentry as $network => $candidate) {
+			$connector = null;
+			$valid = false;
+	
+			// get profile-id
+			switch (strtolower($network)) {
+				case "facebook":
+					$candidate = basename($candidate);
+					if (!ctype_digit($candidate)) {
+						// TODO: determine facebook profile id from username
+						break;
+					}
+					$valid = true;
+					break;
+			}
+			if ($valid) {
+				// build connector
+				switch (strtolower($network)) {
+					case "facebook":
+						switch ($type) {
+							case "avatar":
+								$connector = "https://graph.facebook.com/" . ($candidate) . "/picture?width=720";
+								break;
+							default:
+								break;
+						}
+						break;
+				}
+				
+				// return first valid connector:
+				if ($connector) { return ($connector); }
+			}
+		}
+		
+		// reached only if no valid connectors found
+		return null;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Retrieves social profile data for a contact
+	 */
+	public function fetch(string $addressbookId, string $contactId, string $type) : JSONResponse {
+
+		$url = null;
+		$response = new JSONResponse(array());
+		$response->setStatus(404);
+
+		try {
+			// get corresponding addressbook
+			$addressBooks = $this->manager->getUserAddressBooks();
+			$addressBook = null;
+			foreach($addressBooks as $ab) {
+				if ($ab->getUri() === $addressbookId) {
+					$addressBook = $ab;
+				}
+			}
+			if (is_null($addressBook)) {
+				$response->setStatus(500);
+				return $response; 
+			}
+			
+			// search contact in that addressbook
+			$contact = $addressBook->search($contactId, ['UID'], [])[0];
+			if (is_null($addressBook)) {
+				$response->setStatus(500);
+				return $response; 
+			}
+
+			// get social data
+			$socialprofile = $contact['X-SOCIALPROFILE'];
+			if (is_null($addressBook)) {
+				$response->setStatus(500);
+				return $response; 
+			}
+
+			// retrieve data
+			try {
+				$url = $this->getSocialConnector($socialprofile, $type);
+			}
+			catch (Exception $e) {
+				$response->setStatus(400);
+				return $response;
+			}
+
+			if (empty($url)) {
+				$response->setStatus(501);
+				return $response;
+			}
+
+			$host = parse_url($url);
+			if (!$host) {
+				$response->setStatus(400);
+				return $response;
+			}
+			$opts = [
+				"http" => [
+					"method" => "GET",
+					"header" => "User-Agent: Nextcloud Contacts App"
+				]
+			];
+			$context = stream_context_create($opts);
+			$socialdata = file_get_contents($url, false, $context);
+			if (!$socialdata) {
+				$response->setStatus(404);
+				return $response;
```

Comment:
return new JSONResponse(['message' => 'Could not find ressource'], Http::STATUS_NOT_FOUND);